setup.py

```diff
@@ -50,5 +50,6 @@
     ],
     extras_require={
         "plotting": ["seaborn", "plotly"],
+        "qhsri": ["pymoo", "cvxpy"],
```

Comment:
To ensure the tests and notebooks always run with the same version of pymoo and cvxpy (and their dependencies) you should also update `generate_constraints.sh` to do `pip install -e .[qhsri]` on line 31. Rerunning the script will then regenerate the constraints.txt files for the tests and notebooks so that they include these libraries.